src/include/sof/audio/selector.h

```diff
@@ -42,6 +42,14 @@ struct comp_dev;
 
 /** \brief Maximum supported channel count on output. */
 #define SEL_SINK_CHANNELS_MAX   8
+
+/*
+ * The selector module only needs the output pin format from base config extension,
+ * the input pin format is from base config.
+ */
+#define SEL_NUM_IN_PIN_FMTS	0
+#define SEL_NUM_OUT_PIN_FMTS	1
```

Comment:
@aiChaoSONG, actually, this is not good, the extension will have format for all pins, so the nb_input_pins will be 1.

Reply:
@aiChaoSONG thats right, if a pin uses pin 0 on input and output, the extension will have to contain both formats> We can't selectively avoid sending it with the extension because it is already in the base config. Its easier to just say that all used pins will have formats set in the extension irrespective of what the base config contains